pkg/controller/controller.go

```diff
@@ -248,15 +248,32 @@ func (c *Controller) GetClusters(name string) ([]*model.Cluster, error) {
 	return cl.GetClusters(name)
 }
 
-// DeleteNodePool deletes a node pool
-func (c *Controller) DeleteNodePool(clusterName, name string) error {
+// DeleteCluster deletes a node pool.
+func (c *Controller) DeleteCluster(name string) error {
+	cl, impl := c.Cloud.Clusters()
+	if !impl {
+		return ErrNotImplemented
```

Comment:
Yeah, I agree. It should come from `cloudprovider.ErrNotImplemented`, I'll refactor that, because we have it in other places.